src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/FileWriter.java

```diff
@@ -86,6 +98,12 @@ public FileWriter(String basePath,
         // If we failed on flush we want to throw the error from the put() flow.
         flushError = null;
         this.format = format;
+        this.fileCountOnStage = metricRegistry.counter(KustoKafkaMetricsUtil.constructMetricName(tpname, KustoKafkaMetricsUtil.FILE_COUNT_SUB_DOMAIN, KustoKafkaMetricsUtil.FILE_COUNT_ON_STAGE));

```

Comment:
may be this part can be refactored to introduce a method called

private void initializeMetrics(String tpname, MetricRegistry metricRegistry) {
    this.fileCountOnStage = metricRegistry.counter(KustoKafkaMetricsUtil.constructMetricName(tpname, KustoKafkaMetricsUtil.FILE_COUNT_SUB_DOMAIN, KustoKafkaMetricsUtil.FILE_COUNT_ON_STAGE));
// populate other metrics data
}

and this is invoked from the constructor as 
  initializeMetrics(tpname, metricRegistry);

this refactoring will help keep the constructor clean and delegate the metric initialization to a separate method